### DISCUSSION TITLE
responsive ui？

### DISCUSSION THREAD

**Discussion by atticus-lv** [zauberzeug/nicegui#3006] [Ideas / Feature Requests]
i found a amazing lib to do responsive ui in nicegui 
 https://github.com/CrystalWindSnake/ex4nicegui
the usage of the 'to ref' is really good(just like the auto update in kotlin compose). can it be done in nicegui?

**Comment by falkoschindler** [comment]
Hi @atticus-lv!

The "ex4nicegui" project is definitely impressive! As I wrote in https://github.com/zauberzeug/nicegui/pull/1447#issuecomment-1695740074, integrating it into NiceGUI would probably require changing or duplicating many elements to deal with reactive arguments, complicating the existing implementation even more. Therefore we think it's better to keep it separated as its own extension library. If you like it, feel free to use (and support) it. 🙂
